zfit/core/parameter.py

```diff
@@ -465,23 +465,51 @@ class ComposedParameter(BaseComposedParameter):
     # TODO: raise error if eager is on (because it's very errorprone)
     def __init__(self, name, tensor, **kwargs):
         tensor = ztf.convert_to_tensor(tensor)
-        independend_params = tf.get_collection("zfit_independent")
-        params = get_dependents(tensor=tensor, candidates=independend_params)
+        independent_params = tf.get_collection("zfit_independent")
+        params = get_dependents(tensor=tensor, candidates=independent_params)
         # params_init_op = [param.initializer for param in params]
         params = {p.name: p for p in params}
         # with tf.control_dependencies(params_init_op):
         super().__init__(params=params, initial_value=tensor, name=name, **kwargs)
 
 
 class ComplexParameter(BaseComposedParameter):
-    def __init__(self, name, initial_value, floating=True, dtype=ztypes.complex, **kwargs):
-        initial_value = tf.cast(initial_value, dtype=dtype)
-        real_value = tf.real(initial_value)
-        real_part = Parameter(name=name + "_real", init_value=real_value, floating=floating, dtype=real_value.dtype)
-        imag_value = tf.imag(initial_value)
-        imag_part = Parameter(name=name + "_imag", init_value=imag_value, floating=floating, dtype=imag_value.dtype)
-        params = {'real': real_part, 'imag': imag_part}
-        super().__init__(params=params, initial_value=initial_value, name=name, **kwargs)
+    def __init__(self, name, initial_value, dtype=ztypes.complex, **kwargs):
+        tensor = ztf.convert_to_tensor(initial_value, dtype=dtype)
+        independent_params = tf.get_collection("zfit_independent")
+        params = get_dependents(tensor=tensor, candidates=independent_params)
+        params = {p.name: p for p in params}
+        super().__init__(name=name, params=params, initial_value=tensor, **kwargs)
+
+    @staticmethod
+    def from_cartesian(name, real, imag, dtype=ztypes.complex, **kwargs):
+        return ComplexParameter(name, tf.cast(tf.complex(real, imag), dtype=dtype))
+
+    @staticmethod
+    def from_polar(name, abs, arg, dtype=ztypes.complex, **kwargs):
```

Comment:
abs -> mod?

so we have name consistency with the attribute and no clash with builtins

Reply:
OK!